qa/logging-config/src/test/java/org/elasticsearch/common/logging/JsonLoggerTests.java

```diff
@@ -486,28 +438,61 @@ private void setupLogging(final String config, final Settings settings) throws I
         LogConfigurator.configure(environment);
     }
 
+
     private Matcher<JsonLogLine> logLine(String type, Level level, String nodeName, String component, String message) {
+        return logLine(mapOfParamsToCheck(type, level, nodeName, component, message));
+    }
+
+    private Map<Function<JsonLogLine, Object>, Object> mapOfParamsToCheck(
+        String type, Level level, String nodeName, String component, String message) {
+        return Map.of(JsonLogLine::getType, type,
+            JsonLogLine::getLevel, level.toString(),
+            JsonLogLine::getNodeName, nodeName,
+            JsonLogLine::getComponent, component,
+            JsonLogLine::getMessage, message);
+    }
+
+    private Matcher<JsonLogLine> logLine(String type, Level level, String nodeName, String component, String message,
+                                         Map<Function<JsonLogLine,Object>, Object> additionalProperties) {
+        Map<Function<JsonLogLine, Object>, Object> map = new HashMap<>();
+        map.putAll(mapOfParamsToCheck(type, level, nodeName, component, message));
+        map.putAll(additionalProperties);
+        return logLine(map);
+    }
+
+    private Matcher<JsonLogLine> logLine(Map<Function<JsonLogLine,Object>, Object> map) {
         return new FeatureMatcher<JsonLogLine, Boolean>(Matchers.is(true), "logLine", "logLine") {
 
             @Override
             protected Boolean featureValueOf(JsonLogLine actual) {
-                return Objects.equals(actual.type(), type) &&
-                    Objects.equals(actual.level(), level.toString()) &&
-                    Objects.equals(actual.nodeName(), nodeName) &&
-                    Objects.equals(actual.component(), component) &&
-                    Objects.equals(actual.message(), message);
+                boolean value = true;
```

Comment:
Would it be clearer to do something like this?
```java
return map.entrySet()
    .stream()
    .allMatch(entry -> Objects.equals(entry.getKey().apply(actual), entry.getValue()));
```

Reply:
agree, stream will look better here